### PR TITLE
Wasmtime Support

### PR DESCRIPTION
**Pull request by cohix**

### CODE REVIEW

File: rwasm/runtime/wasmtime/builder.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import (
+	"github.com/bytecodealliance/wasmtime-go"
+	"github.com/pkg/errors"
+	"github.com/suborbital/reactr/rwasm/moduleref"
+	"github.com/suborbital/reactr/rwasm/runtime"
+)
+
+// WasmTimeBuilder is a Wasmer implementation of the instanceBuilder interface
+type WasmTimeBuilder struct {
```
Comment (ospencer) [inline rwasm/runtime/wasmtime/builder.go:11]: Would probably name this `WasmtimeBuilder`

File: rwasm/runtime/wasmtime/builder.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	inst := &WasmtimeInstance{
+		inst:  *wasmTimeInst,
+		store: store,
+	}
+
+	if _, err := inst.Call("_start"); err != nil {
+		if errors.Is(err, runtime.ErrExportNotFound) {
+			// that's ok, not all modules will have _start
+		} else {
+			return nil, errors.Wrap(err, "failed to call _start")
```
Comment (ospencer) [inline rwasm/runtime/wasmtime/builder.go:54]: Would be helpful if this said something that made it obvious that the `_start` function was defined, just that there was an error executing it. I totally see myself seeing this error and going "well does my module even have a _start??"

Maybe something like this:
```suggestion
			return nil, errors.Wrap(err, "failed to call defined _start function")
```
but I imagine there's a better way to word it.

File: rwasm/runtime/wasmtime/host_function.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+// addHostFns adds a list of host functions to an import object
+func addHostFns(linker *wasmtime.Linker, fns ...runtime.HostFn) {
+	for i := range fns {
+		// we create a copy inside the loop otherwise things get overwritten
+		fn := fns[i]
+
+		params := make([]*wasmtime.ValType, fn.ArgCount)
+		for i := 0; i < fn.ArgCount; i++ {
+			params[i] = i32Type
```
Comment (ospencer) [inline rwasm/runtime/wasmtime/host_function.go:21]: Maybe it's obvious, though I think a comment that explains that all host functions only operate on i32s would be good here.

File: rwasm/runtime/wasmtime/host_function.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+		returns := []*wasmtime.ValType{}
+		if fn.Returns {
+			returns = append(returns, i32Type)
+		}
+
+		fnType := wasmtime.NewFuncType(params, returns)
+
+		// this is reused across the normal and Swift variations of the function
+		wasmTimeFunc := func(_ *wasmtime.Caller, args []wasmtime.Val) ([]wasmtime.Val, *wasmtime.Trap) {
```
Comment (ospencer) [inline rwasm/runtime/wasmtime/host_function.go:32]: Would probably name this `wasmtimeFunc` just to be consistent.

File: rwasm/runtime/wasmtime/wasmtime.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+// Deallocate deallocates memory in the instance
+func (w *WasmtimeInstance) Deallocate(pointer int32, length int) {
+	w.Call("deallocate", pointer, length)
+}
+
+// Close closes the instance
+func (w *WasmtimeInstance) Close() {
+	// figure out how to close
```
Comment (ospencer) [inline rwasm/runtime/wasmtime/wasmtime.go:83]: Worth a TODO or issue maybe?

Comment (cohix) [reply rwasm/runtime/wasmtime/wasmtime.go:83]: Yep, #148

**Review by ospencer — APPROVED** [review]

**Pull request merged**